src/webui/www/private/scripts/progressbar.js

```diff
@@ -36,124 +36,82 @@ window.qBittorrent.ProgressBar ??= (() => {
+            this.#light.style.left = "0";
+            this.#light.style.top = "0";
+            this.#light.style.lineHeight = `${ProgressBar.#styles.height}px`;
+
+            this.attachShadow({ mode: "open" });
+            this.shadowRoot.host.id = this.#id;
+            this.shadowRoot.host.style.display = "block";
+            this.shadowRoot.host.style.border = "1px solid var(--color-border-default)";
+            this.shadowRoot.host.style.boxSizing = "content-box";
+            this.shadowRoot.host.style.height = `${ProgressBar.#styles.height}px`;
+            this.shadowRoot.host.style.position = "relative";
+            this.shadowRoot.host.style.margin = "0 auto";
+            this.shadowRoot.appendChild(this.#dark);
+            this.shadowRoot.appendChild(this.#light);
+
+            this.setValue(value);
+        }
 
-    function ProgressBar_setValue(value) {
-        value = Number(value);
-        if (Number.isNaN(value))
-            value = 0;
-        value = Math.min(Math.max(value, 0), 100);
-        this.vals.value = value;
+        getValue() {
+            return this.#value;
+        }
 
-        const displayedValue = `${window.qBittorrent.Misc.toFixedPointString(value, 1)}%`;
-        this.vals.dark.textContent = displayedValue;
-        this.vals.light.textContent = displayedValue;
+        setValue(value) {
+            value = Number(value);
+            if (Number.isNaN(value))
+                value = 0;
+            this.#value = Math.min(Math.max(value, 0), 100);
 
-        const r = Number(this.vals.width * (value / 100));
-        this.vals.dark.style.clipPath = `inset(0 calc(100% - ${r}px) 0 0)`;
-        this.vals.light.style.clipPath = `inset(0 0 0 ${r}px)`;
-    }
+            const displayedValue = `${window.qBittorrent.Misc.toFixedPointString(this.#value, 1)}%`;
+            this.#dark.textContent = displayedValue;
+            this.#light.textContent = displayedValue;
 
-    function ProgressBar_setWidth(value) {
-        if (this.vals.width !== value) {
-            this.vals.width = value;
-            this.style.width = `${value}px`;
-            this.vals.dark.style.width = `${value}px`;
-            this.vals.light.style.width = `${value}px`;
-            this.setValue(this.vals.value);
+            this.#dark.style.clipPath = `inset(0 ${100 - this.#value}% 0 0)`;
+            this.#light.style.clipPath = `inset(0 0 0 ${this.#value}%)`;
         }
     }
 
-    const ProgressBar_checkForParent = (id) => {
-        const obj = document.getElementById(id);
-        if (!obj)
-            return;
-        if (!obj.parentNode)
-            return setTimeout(ProgressBar_checkForParent, 100, id);
-        obj.style.width = "100%";
-        const w = obj.offsetWidth;
-        obj.vals.dark.style.width = `${w}px`;
-        obj.vals.light.style.width = `${w}px`;
-        obj.vals.width = w;
-        obj.setValue(obj.vals.value);
-    };
+    customElements.define("progress-bar", ProgressBar);
```

Comment:
AFAIK currently it being a custom element doesn't bring any benefits and it isn't a requirement. Then it should not be a custom element.

Reply:
OK, I played with it and see the point. Let's keep the custom element.